test/build/config-format/typescript-esnext/typescript.test.js

```diff
@@ -0,0 +1,31 @@
+// eslint-disable-next-line node/no-unpublished-require
+const { run, isWebpack5 } = require('../../../utils/test-utils');
+const { existsSync } = require('fs');
+const { resolve } = require('path');
+
+describe('webpack cli', () => {
+    it('should support typescript esnext file', async () => {
+        const isMacOS = process.platform === 'darwin';
+        const majorNodeVersion = process.version.slice(1, 3);
+        if (majorNodeVersion < 14) {
+            expect(true).toBe(true);
+
+            return;
+        }
+
+        if (isMacOS && !isWebpack5) {
```

Comment:
Why here `isWebpack5`? Happens only for v4 on macos?

Reply:
it's `!isWebpack5` i.e 4 😄

Reply:
yes

Reply:
which is very weird as I don't know how webpack version change is causing this

Reply:
Doesn't work locally either 😞

Reply:
> Doesn't work locally either disappointed

Do you mean it freezes on macos locally too?

Reply:
yep

Reply:
Very weird, bug we will drop webpack v4 in near future, so let's merge